pkg/pod/wait.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+// WaitExited waits for a pod to (run and) exit.
+// This method blocks indefinitely and refreshes the pod state.
+func (p *Pod) WaitExited() error {
+	// isExited implies isExitedGarbage.
+	for !p.isExited && !p.isAbortedPrepare && !p.isGarbage && !p.isGone {
+		// this blocks (waits) as long as the pod is locked, i.e. in pepare, run, exitedGarbage, garbage state
+		if err := p.SharedLock(); err != nil {
+			return err
+		}
+
+		// unlock immediately
+		if err := p.Unlock(); err != nil {
+			return err
+		}
+
+		if err := p.refreshState(); err != nil {
+			return err
+		}
+
+		// if we're in the gap between preparing and running in a split prepare/run-prepared usage, take a nap
+		if p.isPrepared {
+			time.Sleep(time.Second)
+		}
+	}
+
+	// TODO(vc): return error or let caller detect the !p.isExited possibilities?
```

Comment:
I would rather nix this TODO and just document in the function comment that the pod will be in a terminal state if there is no error, but it's up to the caller to determine what that state is.